Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,946 @@
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Text;
+#if !NO_SYSTEM_WEB_DEPENDENCY && !MOBILE && false
+using System.Web.Configuration;
+#endif
+
+namespace System.Web.Util
+{
+    public class HttpEncoder
+    {
+        private static char[] hexChars = "0123456789abcdef".ToCharArray();
+        private static object entitiesLock = new object();
+        private static SortedDictionary<string, char> entities;
+        private static Lazy<HttpEncoder> defaultEncoder;
+        private static Lazy<HttpEncoder> currentEncoderLazy;
+        private static HttpEncoder currentEncoder;
+
+        private static IDictionary<string, char> Entities
+        {
+            get
+            {
+                lock (entitiesLock)
+                {
+                    if (entities == null)
+                    {
+                        InitEntities();
+                    }
+
+                    return entities;
+                }
+            }
+        }
+
+        public static HttpEncoder Current
+        {
+            get
+            {
+                if (currentEncoder == null)
+                {
+                    currentEncoder = currentEncoderLazy.Value;
+                }
+
+                return currentEncoder;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                currentEncoder = value;
+            }
+        }
+
+        public static HttpEncoder Default
+        {
+            get
+            {
+                return defaultEncoder.Value;
+            }
+        }
+
+        private static HttpEncoder()
+        {
+            defaultEncoder = new Lazy<HttpEncoder>(() => new HttpEncoder());
+            currentEncoderLazy = new Lazy<HttpEncoder>(new Func<HttpEncoder>(GetCustomEncoderFromConfig));
+        }
+
+        public HttpEncoder()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this redundant constructor.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)